src/Model/Subscription/SubscriptionBuilder.php

```diff
@@ -127,25 +130,31 @@ public function withDiscount(Discount $discount): self
 
     public function build(): Subscription
     {
-        $subscription = (new Subscription($this->get('id')))
-            ->setStartDate($this->get('startDate'))
-            ->setPrice($this->get('price'))
-            ->setStatus($this->get('status'))
-            ->setCustomer($this->get('customer'))
-            ->setPaymentMethod($this->get('paymentMethod'))
-            ->setBalance($this->get('balance'));
-
-        $addOns = $this->get('addOns') ?? [];
+        $subscription = (new Subscription($this->getId()))
+            ->setStartDate($this->data['startDate'] ?? new DateTime('now', new DateTimeZone('UTC')))
+            ->setPrice($this->data['price'] ?? new NullPrice())
+            ->setStatus($this->data['status'] ?? Status::initialized())
+            ->setCustomer($this->data['customer'] ?? new Customer());
+
+        if (isset($this->data['paymentMethod'])) {
+            $subscription->setPaymentMethod($this->data['paymentMethod']);
+        }
+
+        if (isset($this->data['balance'])) {
+            $subscription->setBalance($this->data['balance']);
+        }
```

Comment:
just for fun... i believe these could become...

```php
isset($this->data['paymentMethod'])) && $subscription->setPaymentMethod($this->data['paymentMethod']);
isset($this->data['balance'])) && $subscription->setBalance($this->data['balance']);
```


![Screenshot_20201208_114535](https://user-images.githubusercontent.com/18272064/101514116-ece92e80-394a-11eb-8193-6f9e012456dc.png)

Reply:
Did PHP become JavaScript? DID I MISS A MEMO?

Reply:
Personally, I am ok trading verbosity for readability here. Cool if I leave as is?

Reply:
totally fine as is... and probably preferred... the bash stuff i've been doing lately has been nice doing checks like that... it's more valuable there.